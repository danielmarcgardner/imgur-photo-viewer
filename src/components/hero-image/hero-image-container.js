import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import HeroImage from './hero-image';

const mapStateToProps = (state, ownProps) => {
  const id = ownProps.match.params.id;
  const heroImage = state.searchResults.images.filter(image => image.id === id);
  return {
    heroImage: heroImage[0],
  };
};

export default connect(mapStateToProps)(withRouter(HeroImage));
