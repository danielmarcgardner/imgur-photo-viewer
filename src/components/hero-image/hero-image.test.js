import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HeroImage from './hero-image';
import { mapStateToProps } from './hero-image-container';

configure({ adapter: new Adapter() });


const heroImage = {
  image: 'https://i.imgur.com/fZ8JNw4.jpg',
  title: 'Skateboarding Bulldog',
  tags: ['Funny', 'Dog', 'Action'],
  score: 100,
  datetime: 'Mon Oct 02 2017',
  id: 'fZ8JNw4',
};
const history = {
  push: jest.fn(),
};

describe('HeroImage', () => {
  it('Renders to page', () => {
    const component = shallow(
      <HeroImage
        heroImage={heroImage}
        history={history}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders to page with no tags', () => {
    const heroImageAlt = {
      image: 'https://i.imgur.com/fZ8JNw4.jpg',
      title: 'Skateboarding Bulldog',
      tags: [],
      score: 100,
      datetime: 'Mon Oct 02 2017',
    };
    const component = shallow(
      <HeroImage
        heroImage={heroImageAlt}
        history={history}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders with all correct classes', () => {
    const component = shallow(
      <HeroImage
        heroImage={heroImage}
        history={history}
      />,
    );
    expect(component.find('#HeroImageContainer').length).toBe(1);
    expect(component.find('#HeroImageHeader').length).toBe(1);
    expect(component.find('#HeroImage').length).toBe(1);
    expect(component.find('#HeroTitle').length).toBe(1);
  });
  it('When button is clicked history.push is called', () => {
    const component = mount(
      <HeroImage
        heroImage={heroImage}
        history={history}
      />,
    );
    component.find('button').simulate('click');
    expect(history.push).toHaveBeenCalled();
    expect(history.push.mock.calls[0][0]).toBe('/');
  });
  it('MapStateToProps returns the proper items', () => {
    const state = {
      searchResults: {
        images: [
          {
            image: 'https://i.imgur.com/fZ8JNw4.jpg',
            title: 'Skateboarding Bulldog',
            tags: ['Funny', 'Dog', 'Action'],
            score: 100,
            datetime: 'Mon Oct 02 2017',
            id: 'fZ8JNw4',
          },
          {
            id: 'gzC6hl4',
            image: 'https://i.imgur.com/gzC6hl4.gif',
            title: 'Tired Bulldog Dad Needs Break From His Playful Puppies.',
            score: 39,
            tags: [],
            datetime: 'Mon Oct 02 2017',
          }, {
            id: 'xE7Yu2v',
            image: 'https://i.imgur.com/xE7Yu2v.jpg',
            title: 'Trevor Donovan with his bulldog Tito and German Shepherd Dogbert.',
            score: 26,
            tags: ['dogs', 'dog', 'german sheâ€¦', 'pets'],
            datetime: 'Mon Oct 02 2017' },
        ],
      },
    };

    const ownProps = {
      match: {
        params: {
          id: 'fZ8JNw4',
        },
      },
    };

    const expected = {
      heroImage: {
        image: 'https://i.imgur.com/fZ8JNw4.jpg',
        title: 'Skateboarding Bulldog',
        tags: ['Funny', 'Dog', 'Action'],
        score: 100,
        datetime: 'Mon Oct 02 2017',
        id: 'fZ8JNw4',
      } };
    const actual = mapStateToProps(state, ownProps);
    expect(actual).toEqual(expected);
  });
});
