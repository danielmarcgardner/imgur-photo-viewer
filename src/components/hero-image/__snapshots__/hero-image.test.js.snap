// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImage Renders to page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeroImage
    heroImage={
        Object {
            "datetime": "Mon Oct 02 2017",
            "id": "fZ8JNw4",
            "image": "https://i.imgur.com/fZ8JNw4.jpg",
            "score": 100,
            "tags": Array [
              "Funny",
              "Dog",
              "Action",
            ],
            "title": "Skateboarding Bulldog",
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="HeroImageContainer"
>
          <div
                    id="HeroImageHeader"
          >
                    <img
                              alt="HeroImage"
                              id="HeroImage"
                              src="https://i.imgur.com/fZ8JNw4.jpg"
                    />
          </div>
          <p
                    id="HeroTitle"
          >
                    Skateboarding Bulldog
          </p>
          <p>
                    Tags: Funny, Dog, Action
          </p>
          <p>
                    Score: 100
          </p>
          <p>
                    Date: Mon Oct 02 2017
          </p>
</div>,
        <button
          onClick={[Function]}
>
          Back To Image Gallery
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="HeroImageHeader"
>
              <img
                            alt="HeroImage"
                            id="HeroImage"
                            src="https://i.imgur.com/fZ8JNw4.jpg"
              />
</div>,
            <p
              id="HeroTitle"
>
              Skateboarding Bulldog
</p>,
            <p>
              Tags: Funny, Dog, Action
</p>,
            <p>
              Score: 100
</p>,
            <p>
              Date: Mon Oct 02 2017
</p>,
          ],
          "id": "HeroImageContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="HeroImage"
                id="HeroImage"
                src="https://i.imgur.com/fZ8JNw4.jpg"
/>,
              "id": "HeroImageHeader",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "alt": "HeroImage",
                "id": "HeroImage",
                "src": "https://i.imgur.com/fZ8JNw4.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Skateboarding Bulldog",
              "id": "HeroTitle",
            },
            "ref": null,
            "rendered": "Skateboarding Bulldog",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Tags: Funny, Dog, Action",
            },
            "ref": null,
            "rendered": "Tags: Funny, Dog, Action",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Score: 100",
            },
            "ref": null,
            "rendered": "Score: 100",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Date: Mon Oct 02 2017",
            },
            "ref": null,
            "rendered": "Date: Mon Oct 02 2017",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Back To Image Gallery",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back To Image Gallery",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="HeroImageContainer"
>
            <div
                        id="HeroImageHeader"
            >
                        <img
                                    alt="HeroImage"
                                    id="HeroImage"
                                    src="https://i.imgur.com/fZ8JNw4.jpg"
                        />
            </div>
            <p
                        id="HeroTitle"
            >
                        Skateboarding Bulldog
            </p>
            <p>
                        Tags: Funny, Dog, Action
            </p>
            <p>
                        Score: 100
            </p>
            <p>
                        Date: Mon Oct 02 2017
            </p>
</div>,
          <button
            onClick={[Function]}
>
            Back To Image Gallery
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="HeroImageHeader"
>
                <img
                                alt="HeroImage"
                                id="HeroImage"
                                src="https://i.imgur.com/fZ8JNw4.jpg"
                />
</div>,
              <p
                id="HeroTitle"
>
                Skateboarding Bulldog
</p>,
              <p>
                Tags: Funny, Dog, Action
</p>,
              <p>
                Score: 100
</p>,
              <p>
                Date: Mon Oct 02 2017
</p>,
            ],
            "id": "HeroImageContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="HeroImage"
                  id="HeroImage"
                  src="https://i.imgur.com/fZ8JNw4.jpg"
/>,
                "id": "HeroImageHeader",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "alt": "HeroImage",
                  "id": "HeroImage",
                  "src": "https://i.imgur.com/fZ8JNw4.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Skateboarding Bulldog",
                "id": "HeroTitle",
              },
              "ref": null,
              "rendered": "Skateboarding Bulldog",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Tags: Funny, Dog, Action",
              },
              "ref": null,
              "rendered": "Tags: Funny, Dog, Action",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Score: 100",
              },
              "ref": null,
              "rendered": "Score: 100",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Date: Mon Oct 02 2017",
              },
              "ref": null,
              "rendered": "Date: Mon Oct 02 2017",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Back To Image Gallery",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back To Image Gallery",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HeroImage Renders to page with no tags 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeroImage
    heroImage={
        Object {
            "datetime": "Mon Oct 02 2017",
            "image": "https://i.imgur.com/fZ8JNw4.jpg",
            "score": 100,
            "tags": Array [],
            "title": "Skateboarding Bulldog",
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="HeroImageContainer"
>
          <div
                    id="HeroImageHeader"
          >
                    <img
                              alt="HeroImage"
                              id="HeroImage"
                              src="https://i.imgur.com/fZ8JNw4.jpg"
                    />
          </div>
          <p
                    id="HeroTitle"
          >
                    Skateboarding Bulldog
          </p>
          <p>
                    Tags: None :(
          </p>
          <p>
                    Score: 100
          </p>
          <p>
                    Date: Mon Oct 02 2017
          </p>
</div>,
        <button
          onClick={[Function]}
>
          Back To Image Gallery
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="HeroImageHeader"
>
              <img
                            alt="HeroImage"
                            id="HeroImage"
                            src="https://i.imgur.com/fZ8JNw4.jpg"
              />
</div>,
            <p
              id="HeroTitle"
>
              Skateboarding Bulldog
</p>,
            <p>
              Tags: None :(
</p>,
            <p>
              Score: 100
</p>,
            <p>
              Date: Mon Oct 02 2017
</p>,
          ],
          "id": "HeroImageContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="HeroImage"
                id="HeroImage"
                src="https://i.imgur.com/fZ8JNw4.jpg"
/>,
              "id": "HeroImageHeader",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "alt": "HeroImage",
                "id": "HeroImage",
                "src": "https://i.imgur.com/fZ8JNw4.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Skateboarding Bulldog",
              "id": "HeroTitle",
            },
            "ref": null,
            "rendered": "Skateboarding Bulldog",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Tags: None :(",
            },
            "ref": null,
            "rendered": "Tags: None :(",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Score: 100",
            },
            "ref": null,
            "rendered": "Score: 100",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Date: Mon Oct 02 2017",
            },
            "ref": null,
            "rendered": "Date: Mon Oct 02 2017",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Back To Image Gallery",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back To Image Gallery",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="HeroImageContainer"
>
            <div
                        id="HeroImageHeader"
            >
                        <img
                                    alt="HeroImage"
                                    id="HeroImage"
                                    src="https://i.imgur.com/fZ8JNw4.jpg"
                        />
            </div>
            <p
                        id="HeroTitle"
            >
                        Skateboarding Bulldog
            </p>
            <p>
                        Tags: None :(
            </p>
            <p>
                        Score: 100
            </p>
            <p>
                        Date: Mon Oct 02 2017
            </p>
</div>,
          <button
            onClick={[Function]}
>
            Back To Image Gallery
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="HeroImageHeader"
>
                <img
                                alt="HeroImage"
                                id="HeroImage"
                                src="https://i.imgur.com/fZ8JNw4.jpg"
                />
</div>,
              <p
                id="HeroTitle"
>
                Skateboarding Bulldog
</p>,
              <p>
                Tags: None :(
</p>,
              <p>
                Score: 100
</p>,
              <p>
                Date: Mon Oct 02 2017
</p>,
            ],
            "id": "HeroImageContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="HeroImage"
                  id="HeroImage"
                  src="https://i.imgur.com/fZ8JNw4.jpg"
/>,
                "id": "HeroImageHeader",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "alt": "HeroImage",
                  "id": "HeroImage",
                  "src": "https://i.imgur.com/fZ8JNw4.jpg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Skateboarding Bulldog",
                "id": "HeroTitle",
              },
              "ref": null,
              "rendered": "Skateboarding Bulldog",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Tags: None :(",
              },
              "ref": null,
              "rendered": "Tags: None :(",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Score: 100",
              },
              "ref": null,
              "rendered": "Score: 100",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Date: Mon Oct 02 2017",
              },
              "ref": null,
              "rendered": "Date: Mon Oct 02 2017",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Back To Image Gallery",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back To Image Gallery",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
